app/lib/methods/handleBase64.ts

```diff
@@ -0,0 +1,21 @@
+export const imageBase64RegExp = new RegExp(/^data:image\/[bmp,gif,ico,jpg,png,svg,webp,x\-icon,svg+xml]+;base64,/);
+const regExpOf120Characters = new RegExp(/^data:image\/[bmp,gif,ico,jpg,png,svg,webp,x\-icon,svg+xml]+;base64,([\w+\/=]{1,120})/);
+const mimeTypeBase64RegExp = new RegExp(/^data:(.+);base64,/);
+
+export function isImageBase64(data: string): boolean {
+	return !!data && imageBase64RegExp.test(data);
+}
+
+export function valueOfFirst120CharactersOfImageBase64(imageBase64: string): string | undefined {
```

Comment:
I don't think you need this function, you just use it to set the base64 name, a New Date() already solves the problem and is less expensive.

Reply:
This would also make your PR simpler and easier to understand, as it would remove two regex and several tests.

Reply:
But if I create a New Date() only, every time the image component renders on the view, will download new data, because we can't track properly his filename or path

Reply:
A long time ago I discussed this with @diegolmello and he asked me to not work with messageId.